### PR TITLE
feat:  EC Hero Banner

### PR DESCRIPTION
**Pull request by nimek2**
# PR description

Please drop a few lines about the PR: what it does, how to test it, etc.

## QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [x] I have put the vanilla component as `devDependencies`
- [x] I have put the specs package as `devDependencies`
- [x] I have added the components directly used in the twig file (with `include` or `embed`) as `dependencies`
- [x] My component is listed in `@ecl-twig/ec-components`'s `dependencies`
- [x] My variables naming follow the guidelines (snake case for twig)
- [x] I have provided tests
- [x] I have provided documentation (for the "notes" tab)
- [x] If my local `yarn.lock` contains changes, I have committed it
- [x] I have given my PR the proper label (`pr: review needed` to indicate that I'm done and now waiting for a review ,`pr: wip` to indicate that I'm actively working on it ...)

### CODE REVIEW

**Comment by hapertown** [comment]
@emeryro 
Could you see the [test results](https://drone.fpfis.eu/ec-europa/ecl-twig/415/9)? I guess there is a problem with the syntax of the file, but I do not know how to find the cause...

File: src/ec/packages/ec-component-hero-banner/hero-banner.html.twig [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+{% endif %}
+
+{% if extra_attributes is defined and extra_attributes is not empty and extra_attributes is iterable %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %}
+
+{# Print the result #}
+{% spaceless %}
```
Comment (emeryro) [inline src/ec/packages/ec-component-hero-banner/hero-banner.html.twig]: As you may have seen on slack, we had a lots of discussion concerning the best way to handle whitespaces.
We ended up with a [set of rules](https://github.com/ec-europa/ecl-twig/blob/5aadb4708da7e5fe27d7bbb7b8b1d91fae4adacc/docs/decisions/001-strip-whitespaces.md).
Please have a look at it.
Basically the only changes that have to be done here is to put the spaceless at the top of the file, and remove dashes for inline twig tags

File: src/ec/packages/ec-component-hero-banner/hero-banner.story.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  if ('image' in b) {
+    banner.image = 'https://v2--europa-component-library.netlify.com'.concat(
+      b.image
+    );
+  }
+
+  return banner;
+}
+
+storiesOf('Components/HeroBanner', module)
```
Comment (emeryro) [inline src/ec/packages/ec-component-hero-banner/hero-banner.story.js]: please add a space between Hero and Banner

File: src/ec/packages/ec-component-hero-banner/hero-banner.html.twig (hunk 1 above)
Comment (hapertown) [reply src/ec/packages/ec-component-hero-banner/hero-banner.html.twig]: According to the documentation, now (https://github.com/ec-europa/ecl-twig/pull/49/commits/b61ae00e3ae1e87c10ce6b1bddf1950e0fae7785) it should be good - all tags placed inside (and not between) should be removed `-`, but where I have conditional instructions or loading another component, the `-` should stay?

File: src/ec/packages/ec-component-hero-banner/hero-banner.story.js (hunk 2 above)
Comment (hapertown) [reply src/ec/packages/ec-component-hero-banner/hero-banner.story.js]: Fixed in https://github.com/ec-europa/ecl-twig/pull/49/commits/b61ae00e3ae1e87c10ce6b1bddf1950e0fae7785

File: src/ec/packages/ec-component-hero-banner/hero-banner.html.twig [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+{# variables #}
+{% set _type = type|default('default') %}
+{% set _centered = centered|default(true) %}
+{% set _title = title|default('') %}
+{% set _description = description|default('') %}
+{% set _image = image|default('') %}
+
+{% set _button = {
+  variant: 'call',
+  icon: {
```
Comment (emeryro) [inline src/ec/packages/ec-component-hero-banner/hero-banner.html.twig]: the icon should come from the data, not from the template. Icon may vary depending on context (or may not be present at all)

so there is no need to set value for it there

File: src/ec/packages/ec-component-hero-banner/hero-banner.html.twig [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  label: '',
+  extra_classes: 'ecl-hero-banner__button'
+  }
+%}
+
+{% set _css_class = ['ecl-hero-banner ecl-hero-banner--'~_type]|join(' ') %}
+{% set _extra_attributes = '' %}
+
+{# Internal logic - Process properties #}
+{% if button is defined and button is not empty %}
```
Comment (emeryro) [inline src/ec/packages/ec-component-hero-banner/hero-banner.html.twig]: why do you need to do these modifications on button data?
As far as I see, the data is already set in the correct format in your story.js file, so you should just have to use it as is.

File: src/ec/packages/ec-component-hero-banner/package.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  "description": "ECL Twig - EC Hero Banner",
+  "publishConfig": {
+    "access": "public"
+  },
+  "dependencies": {
+    "@ecl-twig/ec-component-button": "^0.0.1-alpha"
+  },
+  "devDependencies": {
+    "@ecl/ec-resources-icons": "~2.1.0",
+    "@ecl/ec-specs-hero-banner": "~2.1.0",
```
Comment (emeryro) [inline src/ec/packages/ec-component-hero-banner/package.json]: please use the same version for all ECL packages (`^2.1.1`)

File: src/ec/packages/ec-component-hero-banner/hero-banner.html.twig (hunk 4 above)
Comment (hapertown) [reply src/ec/packages/ec-component-hero-banner/hero-banner.html.twig]: I thought that some button parameters will be constant - I wanted to give the user the ability to change only those parameters that are necessary. But in this case, what you write, everything will be defined directly by the user, I also removed the unnecessary code.
Fixed in https://github.com/ec-europa/ecl-twig/pull/49/commits/97024e465b8650a3f7608ed55c200d09723e45b6

File: src/ec/packages/ec-component-hero-banner/hero-banner.html.twig (hunk 3 above)
Comment (hapertown) [reply src/ec/packages/ec-component-hero-banner/hero-banner.html.twig]: Check my answer https://github.com/ec-europa/ecl-twig/pull/49#discussion_r262992520

File: src/ec/packages/ec-component-hero-banner/package.json (hunk 5 above)
Comment (hapertown) [reply src/ec/packages/ec-component-hero-banner/package.json]: Updated in https://github.com/ec-europa/ecl-twig/pull/49#discussion_r262992520

**Pull request merged**